STYLE.md

```diff
@@ -0,0 +1,336 @@
+# Dafny Standard Library Style Guide
+
+This style guide provides coding conventions for the Dafny Standard Library code.
+
+*This documentation is a modification of the Dafny Style Guide and is still in progress.*
+
+## Naming Convention
+
+Any **classes**, **modules**, **datatypes**, and **newtypes** are named with `PascalCase`.
+
+    Module MyModule {
+        ...
+    }
+    
+Any **lemmas**, **predicates**, **functions**, **methods** and **variables** are named with `snake_case`.
```

Comment:
I guess we interpreted that style guide as internal to Dafny, so we decided to create something fresh for the standard library.  Perhaps we can consider this a strong vote for snake case.

Reply:
Yeah, that style guide has definitely not been aggressively or consistently applied AFAICT. I don't think it was intended to be internal to the Dafny implementation - there's been no effort to standardize all the regression test cases or anything.

@RustanLeino you may want to break the tie here on what you intended. :)

Regardless of the outcome, perhaps we could build a simple refactoring utility to convert identifiers from one style to another, since no matter the outcome we'll have a fair bit of existing Dafny code that will want to migrate styles.

Reply:
Yes, a refactoring utility, or some equivalent of `rust-fmt` would be quite nice.

Reply:
I confirmed with @RustanLeino offline that he did intend PascalCase as the standard. I also spotted another data point in the built-in datatypes: `iterator` classes automatically define a `MoveNext()` method. :)

Reply:
Since you and Rustan feel that strongly about PascalCase, @sarahc7 has kindly offered to update all of the library code.  One plus is that we can now use `Map` instead of `Apply`, since we will no longer collide with the `map` keyword.

Reply:
Thanks for humouring us, and thank you @sarahc7! :)

Reply:
The library code is now updated to use PascalCase!